src/extensions/UserDefinedFormControllerExtension.php

```diff
@@ -24,6 +25,28 @@ class UserDefinedFormControllerExtension extends Extension
      */
     public function onBeforeInit()
     {
+        /** @var UserDefinedFormController $owner */
+        $owner = $this->owner;
+        $url = $owner->getRequest()->getURL();
+        // Start a clean session if the user visits the original form
+        // This should never run on the 'partial' or 'ping' URL
+        if (strpos($url, 'partial') !== 0 && !strpos($url, 'ping')) {
+            $startNew = true;
+            $session = $owner->getRequest()->getSession();
+            $id = $session->get(PartialSubmissionController::SESSION_KEY);
+            // Check if there is an existing partial submission
+            $existing = PartialFormSubmission::get()->byID((int)$id);
+            if ($existing) {
+                // Check if it has started yet, we need to start a new one, if it has started
+                $startNew = $existing->isStarted();
```

Comment:
If the partial form has started (like if there's a partial form in a different tab), then it will create a new one? 

What do we do with cases like, in one tab you have partial page, in another tab is a new one? How do we manage the sessions?